sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,192 @@
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='hard')
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf2 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft')
+    >>> eclf2 = eclf2.fit(X, y)
+    >>> print(eclf2.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf3 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft', weights=[2,1,1])
+    >>> eclf3 = eclf3.fit(X, y)
+    >>> print(eclf3.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+      
+    """
+    def __init__(self, estimators, voting='hard', weights=None):
+        self.estimators = estimators
+        self.voting = voting
+        self.weights = weights
+
+    
+    def fit(self, X, y):
+        """ 
+        Fits the estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        
+        if self.voting not in ('soft', 'hard'):
+            raise ValueError("voting must be 'soft' or 'hard'; got (voting=%r)"
+                             % voting)
+                                                  
+        if self.weights and len(self.weights) != len(self.estimators):
+            raise ValueError('Number of classifiers and weights must be equal'\
+                         '; got %d weights, %d estimators' 
+                         % (len(self.weights), len(self.estimators)))      
+        
+        self.le_ = LabelEncoder()
+        self.le_.fit(y)
+        self.classes_ = self.le_.classes_
+ 
+        for clf in self.estimators:
+            clf.fit(X, y)
+                   
+        return self
+            
+    
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_samples]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.voting == 'soft':
+            avg = self.predict_proba(X)
+            maj = self.le_.inverse_transform(np.argmax(avg, axis=1))
+        
+        else: # 'hard' voting
+            predictions = self._predict(X)
+
+            maj = np.apply_along_axis(lambda x: np.argmax(np.bincount(x, weights=self.weights)), 
+                    axis=1, arr=predictions)
+        
+        return maj
+            
+    
+    def predict_proba(self, X):
+        """
```

Comment:
The method should have a description in the first line of the docstring (which can be really stupid)